dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -436,262 +436,262 @@ protected void loadPermissions(Contentlet con, Map<String,String> m) throws DotD
+							newRelatedIds.add(parentId);
+							oldRelatedIds.remove(parentId);
+						} else {
+							newRelatedIds.add(childId);
+							oldRelatedIds.remove(childId);
+						}
+					}
+				}
+
+				//Taking the disjunction of both collections will give the old list of dependencies that need to be removed from the
+				//re-indexation and the list of new dependencies no re-indexed yet
+				dependenciesToReindex.addAll(CollectionUtils.disjunction(oldRelatedIds, newRelatedIds));
+			}
+		}
+		return dependenciesToReindex;
 	}
 
 	protected void loadRelationshipFields(Contentlet con, Map<String,String> m) throws DotStateException, DotDataException {
-	    DotConnect db = new DotConnect();
-        db.setSQL("select * from tree where parent = ? or child = ? order by tree_order asc");
-        db.addParam(con.getIdentifier());
-        db.addParam(con.getIdentifier());
+		DotConnect db = new DotConnect();
+		db.setSQL("select * from tree where parent = ? or child = ? order by tree_order asc");
+		db.addParam(con.getIdentifier());
+		db.addParam(con.getIdentifier());
 
-        for(Map<String, Object> relatedEntry : db.loadObjectResults()) {
+		for(Map<String, Object> relatedEntry : db.loadObjectResults()) {
 
-            String childId = relatedEntry.get(ESMappingConstants.CHILD).toString();
-            String parentId = relatedEntry.get(ESMappingConstants.PARENT).toString();
-            String relType=relatedEntry.get(ESMappingConstants.RELATION_TYPE).toString();
-            String order = relatedEntry.get(ESMappingConstants.TREE_ORDER).toString();
+			String childId = relatedEntry.get(ESMappingConstants.CHILD).toString();
+			String parentId = relatedEntry.get(ESMappingConstants.PARENT).toString();
+			String relType=relatedEntry.get(ESMappingConstants.RELATION_TYPE).toString();
+			String order = relatedEntry.get(ESMappingConstants.TREE_ORDER).toString();
 
-            Relationship rel = FactoryLocator.getRelationshipFactory().byTypeValue(relType);
+			Relationship rel = FactoryLocator.getRelationshipFactory().byTypeValue(relType);
 
-            if(rel!=null && InodeUtils.isSet(rel.getInode())) {
-                boolean isSameStructRelationship = rel.getParentStructureInode().equalsIgnoreCase(rel.getChildStructureInode());
+			if(rel!=null && InodeUtils.isSet(rel.getInode())) {
+				boolean isSameStructRelationship = rel.getParentStructureInode().equalsIgnoreCase(rel.getChildStructureInode());
 
-                String propName = isSameStructRelationship ?
-                        (con.getIdentifier().equals(parentId)?rel.getRelationTypeValue() + ESMappingConstants.SUFFIX_CHILD:rel.getRelationTypeValue() + ESMappingConstants.SUFFIX_PARENT)
-                        : rel.getRelationTypeValue();
+				String propName = isSameStructRelationship ?
+						(con.getIdentifier().equals(parentId)?rel.getRelationTypeValue() + ESMappingConstants.SUFFIX_CHILD:rel.getRelationTypeValue() + ESMappingConstants.SUFFIX_PARENT)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 167 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)